src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt

```diff
@@ -187,4 +188,39 @@ data class Licence(
     }
     return false
   }
+
+  fun earliestReleaseDate(listOfBankHolidays: List<LocalDate>, workingDays: Int): LocalDate? {
+    val releaseDate = actualReleaseDate ?: conditionalReleaseDate
+    val possibleReleaseDates: MutableList<LocalDate> = mutableListOf()
+
+    if (releaseDate !== null && isBankHolidayOrWeekend(listOfBankHolidays, releaseDate)) {
+      return getEarliestReleaseDate(listOfBankHolidays, workingDays, releaseDate, possibleReleaseDates)
+    }
+    return releaseDate
+  }
+
+  /** Friday is also considered as weekend */
+  private fun isBankHolidayOrWeekend(listOfBankHolidays: List<LocalDate>, releaseDate: LocalDate?): Boolean {
+    val dayOfWeek = releaseDate?.dayOfWeek
+    if (dayOfWeek == DayOfWeek.FRIDAY || dayOfWeek == DayOfWeek.SATURDAY || dayOfWeek == DayOfWeek.SUNDAY) {
+      return true
+    }
+    return listOfBankHolidays.contains(releaseDate)
+  }
+
+  private fun getEarliestReleaseDate(
```

Comment:
I think it'd make sense to put this on the BankHoliday service. 

in the licence service you could have something like:
```kt
val releaseDate = licence.actualReleaseDate ?: licence.conditionalReleaseDate
val earliestReleaseDate = bankHoliday.getEarliestReleaseDate(releaseDate) 
```